pkg/k8shandler/collection.go

```diff
@@ -334,7 +334,11 @@ func isDaemonsetDifferent(current *apps.DaemonSet, desired *apps.DaemonSet) (*ap
 		logrus.Infof("Collector resource(s) change found, updating %q", current.Name)
 		different = true
 	}
-
+	if !reflect.DeepEqual(current.Spec.Template.Spec.Containers[0].Env, desired.Spec.Template.Spec.Containers[0].Env) {
+		logrus.Infof("Collector container EnvVar change found, updating %q", current.Name)
+		current.Spec.Template.Spec.Containers[0].Env = desired.Spec.Template.Spec.Containers[0].Env
+		different = true
```

Comment:
Does this mean we will no longer be able to set environment variables in the fluentd/rsyslog daemonsets to control the behavior?

Reply:
this means if your operator is in 'managed' it will revert any changes to what is expected.  We will need to scale down our operator or make it unmanaged in order to do any testing unrelated to what the operator does

Reply:
Right - I get that.
This is a big change in behavior.  This means, in 4.1, you could set env. vars. directly in the fluentd daemonset in Managed mode and they would apply.  However, big change in 4.2, if you set env. vars. directly in the fluentd daemonset in Managed mode, they will _not_ apply - they will be reverted.
If this is an intentional behavioral change in our product, we need to document this in the release notes and in the product documentation.

Reply:
It is my opinion after operators were introduced that being able to modify env vars is not the correct behavior.  The whole point of the operator pattern is that we are supposed to know what is best in the operational behavior and we set the things that are necessary and expose knobs in the CR as needed.  The operator owns the resources it creates and has the right to ensure they are to the spec it defines.  The fact that you could tweak the env is really a bug in behavior and not by design.  If you need to tweak it then either you should: go into unmanaged or log an RFE for us to promote something into the CR.

As a side note, the forwarding feature will explicitly rely on an env var that contains a hash of its config and the fact that it changes as the mechanism to ensure pipeline changes are causing the DS to roll out

Reply:
> It is my opinion after operators were introduced that being able to modify env vars is not the correct behavior

I agree, but this is still a big change that may break customer deployments when they upgrade to 4.2.  When you take away features, expect to hear about it . . .